exec/java-exec/src/main/java/org/apache/drill/exec/planner/physical/NestedLoopJoinPrule.java

```diff
@@ -49,7 +49,7 @@ protected boolean checkPreconditions(DrillJoinRel join, RelNode left, RelNode ri
       PlannerSettings settings) {
     JoinRelType type = join.getJoinType();
 
-    if (! (type == JoinRelType.INNER || type == JoinRelType.LEFT)) {
+    if (!(type == JoinRelType.INNER || (type == JoinRelType.LEFT && JoinUtils.hasScalarSubqueryInput(left, right)))) {
```

Comment:
@amansinha100 - @Serhii-Harnyk  and I discussed this.  If we return an error in the case I noted, then we will actually fail a bunch of existing unit tests.  That's because there is Calcite code that flattens a subquery into an equivalent left outer join that uses scalar subqueries.  Those queries end up failing.  Given that, we will end up breaking queries that previously were able to run.  Your thoughts on this?